### DISCUSSION TITLE
Run workflow described in another place

### DISCUSSION THREAD

**Discussion by temamagic** [woodpecker-ci/woodpecker#4525] [Ask a Question]
Hi! I'm trying to solve the following problems:

Push was made to repository A, I need to create a specified workflow descibed in repository B

So I want to control the workflow not in this repository but in another place, so that it is impossible to influence the process or so that I have a single place to configure pipelines for all repositories

I tried to use woodpeckerci/plugin-trigger but I don't understand where I need to get the token from, and is this the right way at all

Maybe we can include external workflows or something?

**Comment by qwerty287** [comment]
Check out the externam configuration api: https://woodpecker-ci.org/docs/administration/advanced/external-configuration-api

**Reply by temamagic** [reply]
Looks like exactly what I want! Do you have an example of validation that is up to date? The one I found in your repository doesn't seem to work, or I don't know what's wrong

https://github.com/yaronf/httpsign/blob/fde4fb287b1e75cc6208bfdd70ae2082c5a2736d/signatures.go#L568

i failed here, this wants header signature-input, but woodpecker didn't send them, and this is not the only error I encounter.

I would be very grateful for an example with http.request (I'm using fiber with valyala/fasthttp and filling http.request manually)

**Reply by qwerty287** [reply]
You can check out https://github.com/woodpecker-ci/example-extensions/blob/main/utils/utils.go.

Please ignore the rest of the repo for now, that's still a WIP as we're currently working on refactoring the API system (see #3349). Or you can checkout out tests, these cases should be covered.

**Comment by anbraten** [comment]
Maybe the trigger plugin is what you are looking for: https://woodpecker-ci.org/plugins/Trigger
